app/scripts/components/common/map/utils.ts

```diff
@@ -123,3 +122,63 @@ export async function requestQuickCache({
   return quickCache.get(key);
 }
 
+type Fn = (...args: any[]) => any;
+
+type ObjResMap<T> = {
+  [K in keyof T]: Res<T[K]>;
+};
+
+type Res<T> = T extends Fn
+  ? T extends DatasetDatumFn<DatasetDatumReturnType>
+    ? DatasetDatumReturnType
+    : never
+  : T extends any[]
+  ? Res<T[number]>[]
+  : T extends object
+  ? ObjResMap<T>
+  : T;
+
+export function resolveConfigFunctions<T>(
+  datum: T,
+  bag: DatasetDatumFnResolverBag
+): Res<T>;
+/* eslint-disable-next-line no-redeclare */
+export function resolveConfigFunctions<T extends any[]>(
+  datum: T,
+  bag: DatasetDatumFnResolverBag
+): Res<T[number]>[];
+/* eslint-disable-next-line no-redeclare */
+export function resolveConfigFunctions(
```

Comment:
TBH I'm struggling to understand what this does. `resolveConfigFunctions` can call itself if datum is iterable, that I get. but then in other situations, it calls datum as a function? How is that related to the call in index.tsx line 220? What does `bag` mean exactly? 

Sorry, might just be a case of Friday afternoon.

Reply:
This basically resolves the functions in the dataset config and is needed to support this behavior.
The [docs](https://github.com/NASA-IMPACT/veda-ui/blob/main/docs/content/frontmatter/layer.md#function-values) show some examples and explain the behavior. If it is not clear, we should improve the docs.

This was used in the past for the customization of the compare map message. Although it is a neat little power user functionality, not sure it is used (known/needed) a lot.